bin/client.js

```diff
@@ -142,20 +141,131 @@ cline.on('--help', () => {
+* @return {string}- String containing the full body data.
+*/
+let _post_parse_body = async () => {
+  let parse_body = new Promise((resolve, reject) => {
+    let temp = "";
+    process.stdin.on('data', ( data ) => {
+      temp += data;
+    });
+    process.stdin.on('end', () => {
+      resolve( temp );
+    });
+    process.stdin.on('error', ( err ) => {
+      reject( err );
+    });
+  });
+  let parsedBody = await parse_body;
+  return parsedBody;
+};
+
+/**
+* Creates a queue of instances of requestWorker with the given uriData
+* and executes them one by one. Once finished, the task resolves.
+* Returns a promise object for the queue.
+* @param {object} uriData - An array of all the uri and any relevant
+*                           information.
+* @param output           - Passthrough stream instance.
+* @param {object} task    - Contains information about the request made.
+* @param callback         - Callback function to enable error catching. (?)
+* @return {object}        - Promise object to guarantee that the queue finishes fully.
+*/
+let _make_tasks_queue = (uriData, output, task, callback) => {
+  let queuePromise = new Promise((resolve, reject) => {
+    let q = asyncModule.queue( requestWorker, 1 );
+
+    q.drain(() => {
+      LOGGER.debug('All items have been processed in internal queue');
+      output.end();
+      resolve();
+    });
+
+    q.pause(); // To prevent run condition adding tasks vs processing queue
+    /* jshint -W083 */
+    // functions within a loop
+    for ( var i = 0; i < uriData.uris.length; i++ ) {
+      let newTask = {
+        uri:     uriData.uris[i],
+        headers: uriData.headers4uris[i]
+      };
+      if ( task.ca ) {
+        newTask.ca = task.ca;
+      }
+      LOGGER.debug('Pushing to queue: ' + JSON.stringify( newTask ));
+      q.push( newTask, ( err ) => {
+        if ( !err ) {
+          LOGGER.debug('Finished task: ' + JSON.stringify( newTask ));
+        } else {
+          callback( err );
+          reject();
+        }
+      });
+    }
+    /* jshint +W083 */
+    q.resume();
+  });
+  return queuePromise;
+};
+
+/**
+* Checks if the request made is done by the POST method. If so, parse the body
+* of the request and apply it to the options. Finally, resolve for either option.
+* @param {boolean} POST - Boolean that's true if the POST option was enabled.
+*                         Otherwise it's undefined.
+* @param {object} options - The options for the request, with .body being modified
+*                           with the input data is POST is true.
+*/
+let _check_for_post = ( is_post, options ) => {
+  return new Promise((resolve, reject) => {
+    if ( is_post ) {
+      options.method = 'POST';
+      post_request = false;
+      try {
+        options.headers["Content-type"] = "application/json";
+        let parsedBody = _post_parse_body();
+        parsedBody.then((body) => {
+          options.body = body;
+          LOGGER.debug('First .then');
```

Comment:
This debug message should contain a better message. "Executing after parsing post body" or something which will help to debug later on